.circleci/config.yml

```diff
@@ -18,21 +47,147 @@ jobs:
+          curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash -
+          sudo apt-get install -y nodejs
+    - run:
+        name: install packages
+        command: npm ci
     - save_cache:
-        key: dependency-cache-{{ checksum "package-lock.json" }}
         paths:
-        - ./node_modules
+          - ~/.npm
+        key: node-cache-{{ checksum "package-lock.json" }}
+    - run:
+        name: Integration tests
+        command: npm run test:integration
+
+  generateDoc:
+    executor: node_executor
+    steps:
+    - checkout
+    - restore_cache:
+        keys:
+          - node-cache-{{ checksum "package-lock.json" }}
+          - node-cache-
+    - run:
+        name: install packages
+        command: npm ci
+    - save_cache:
+        paths:
+          - ~/.npm
+        key: node-cache-{{ checksum "package-lock.json" }}
+    - run:
+        name: build doc
+        command: npm run generate-docs
+    - run:
+        name: archive doc
+        command: tar -czf web3-eea-sdk-doc.tgz ./docs/out/web3-eea
+    - store_artifacts:
+        path: ./web3-eea-sdk-doc.tgz
+    - run:
+        name: organise doc dirs
+        command: node .circleci/organise_doc.js --rootdir=$(pwd) --tag="${CIRCLE_TAG}"
+    - persist_to_workspace:
+        root: ~/project
+        paths:
+          - ./docs
+
+  publishDoc:
+    executor: node_executor
+    steps:
+    - checkout
+    - attach_workspace:
+        at: /tmp/workspace
+    - add_ssh_keys:
+        fingerprints:
+          - 'b4:b3:84:2a:8d:25:45:39:f4:b1:86:4d:d4:db:fb:b4'
     - run:
-        name: test
-        command: npm test
+        name: Set Git user params
+        command: |
+          git config --global user.name $CIRCLE_USERNAME
+          git config --global user.email "${CIRCLE_USERNAME}@users.noreply.github.com"
+    - deploy:
+        name: Commit and push docs changes
+        command: |
+          # publishing happens by pushing files to gh-pages branch
+          readonly BRANCH='test-gh-pages' # TODO change to gh-pages before PR merge
+
+          # check if branch exists on origin and if not exits with error.
+          # the branch should be manually created and protected to be used.
+          git ls-remote --heads --exit-code origin $BRANCH
+          if [ $? != 0 ]; then
+              echo "Error: ${BRANCH} branch doesn't exist on ${CIRCLE_REPOSITORY_URL}"
+              exit 1
+          fi
+
+          # switch to the publishing branch
+          git checkout "$BRANCH"
+
+          # copy the content of the docs folder built on the previous job and attached
+          # using workspace on temp location
+          # cp with T option to override existing content, specially usefull for latest
+          cp -aT /tmp/workspace/docs/. .
+
+          # add a readme to make sure devs understand that this publishing branch is generated
+          # by CI (use template readme) and replace the current readme that is only for source dir
+          rm README.md
```

Comment:
this `rm` isn't necessary. the `mv` command will overwrite existing files.

To test it yourself, try

```
touch file_one
touch file_two
mv file_two file_one
ls file_*
```
This should only show `file_one` in the output of ls.